src/txn/txn.c

```diff
@@ -1092,10 +1105,16 @@ __wt_txn_rollback(WT_SESSION_IMPL *session, const char *cfg[])
 		case WT_TXN_OP_BASIC_ROW:
 		case WT_TXN_OP_INMEM_COL:
 		case WT_TXN_OP_INMEM_ROW:
-			WT_ASSERT(session,
-			    upd->txnid == txn->id ||
-			    upd->txnid == WT_TXN_ABORTED);
-			upd->txnid = WT_TXN_ABORTED;
+			if (prepare) {
+				F_CLR(txn, WT_TXN_PREPARE);
```

Comment:
As transaction's in prepared state are not allowed any other operations apart from commit/ rollback, but txn_op_resolve need to perform a cursor search, the prepare state is cleared temporarily.
Suggestions to restructure are most welcome.